### PR TITLE
Merging Stage to Production

### PR DESCRIPTION
**Pull request by ravinderwalkover**
Merging Stage to Production

### CODE REVIEW

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 1]
```diff
@@ -764,7 +786,7 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
                 if (isElectron) {
                     this.router.navigate(['/login']);
                 } else {
-                    window.location.href = (environment.production) ? `https://giddh.com/login/?action=logout` : `https://test.giddh.com/login/?action=logout`;
+                    window.location.href = (environment.production) ? `https://beta.giddh.com/login/?action=logout` : `https://test.giddh.com/login/?action=logout`;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: Change it back to giddh.com

File: apps/web-giddh/src/app/store/Inventory/inventory.reducer.ts [hunk 2]
```diff
@@ -459,14 +459,8 @@ export function InventoryReducer(state: InventoryState = initialState, action: C
...
-                let activeGroup = _.cloneDeep(state.activeGroup) || updateStockResp.body.stockGroup;
-                let stateActiveGrp: StockGroupResponse = null;
-                let myGrp = removeStockItemAndReturnIt(groupArray, activeGroup.uniqueName, updateStockResp.queryString.stockUniqueName, null);
-                if (myGrp) {
-                    myGrp.name = updateStockResp.body.name;
-                    myGrp.uniqueName = updateStockResp.body.uniqueName;
-                    addStockItemAtIndex(groupArray, activeGroup.uniqueName, myGrp);
-                }
+                let activeGroupUniqueName = _.cloneDeep(updateStockResp.queryString.stockGroupUniqueName);
+                updateStockIteminGroupArray(groupArray,activeGroupUniqueName, updateStockResp);
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/store/Inventory/inventory.reducer.ts:463]: Incorrect camelCase name
```suggestion
                updateStockItemInGroupArray(groupArray,activeGroupUniqueName, updateStockResp);
```

File: apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.scss [hunk 3]
```diff
@@ -96,4 +96,10 @@
             height: 36px;
         }
     }
+     .form-inline.financial-selectYear{
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.scss:99]:
```suggestion
     .form-inline.financial-select-year{
```

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    @Output() public dateSelected: EventEmitter<any> = new EventEmitter<any>();
+
+    /** Internal data model */
+    private innerValue: any = '';
+    /** This is used to show default date */
+    public calendarDate: any = '';
+
+    //Placeholders for the callbacks which are later provided by the Control Value Accessor
+    private onTouchedCallback: () => void = noop;
+    private onChangeCallback: (_: any) => void = noop;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts:38]: Missing JS doc

File: apps/web-giddh/src/environments/environment.prod.ts [hunk 5]
```diff
@@ -8,7 +8,7 @@ enableProdMode();
 export const environment: Environment = {
     production: true,
     showDevModule: false,
-    AppUrl: 'https://app.giddh.com',
+    AppUrl: 'https://beta.giddh.com',
```
Comment (prateek-giddh) [inline apps/web-giddh/src/environments/environment.prod.ts:11]: Please revert this

File: apps/web-giddh/src/environments/environment.stage.ts [hunk 6]
```diff
@@ -9,7 +9,7 @@ export const environment: Environment = {
     production: true,
     showDevModule: false,
     AppUrl: 'https://stage.giddh.com/',
-    ApiUrl: 'https://apitest.giddh.com/',
+    ApiUrl: 'https://api.giddh.com/',
```
Comment (prateek-giddh) [inline apps/web-giddh/src/environments/environment.stage.ts:12]: Should be apirelease

File: apps/web-giddh/src/environments/environment.stage.cordova.ts [hunk 7]
```diff
@@ -9,7 +9,7 @@ export const environment: Environment = {
     production: true,
     showDevModule: false,
     AppUrl: './',
-    ApiUrl: 'https://apitest.giddh.com/',
+    ApiUrl: 'https://api.giddh.com/',
```
Comment (prateek-giddh) [inline apps/web-giddh/src/environments/environment.stage.cordova.ts:12]: Should be apirelease

File: apps/web-giddh/src/environments/environment.ts [hunk 8]
```diff
@@ -9,8 +9,8 @@ Error.stackTraceLimit = Infinity;
 
 export const environment: Environment = {
     production: false,
-    AppUrl: 'http://localhost:3000/',
-    ApiUrl: 'https://apitest.giddh.com/',
+    AppUrl: 'https://beta.giddh.com/',
```
Comment (prateek-giddh) [inline apps/web-giddh/src/environments/environment.ts:12]: Please revert this

File: apps/web-giddh/src/meta/success.html [hunk 9]
```diff
@@ -29,7 +29,7 @@ <h5>You can close this window.</h5>
     </div>
 
     <div class="copy-right">
-        <p>All right are reserved &copy; 2020 Walkover <a href="https://app.giddh.com/" target="_Blank">Giddh.com</a>
+        <p>All right are reserved &copy; 2020 Walkover <a href="https://beta.giddh.com/" target="_Blank">Giddh.com</a>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/meta/success.html:32]: Please revert this

File: apps/web-giddh/src/meta/yodlee-success.html [hunk 10]
```diff
@@ -34,7 +34,7 @@ <h5>You can close this window.</h5>
     </div>
 
     <div class="copy-right">
-        <p>All right are reserved &copy; 2020 Walkover <a href="https://app.giddh.com/" target="_Blank">Giddh.com</a>
+        <p>All right are reserved &copy; 2020 Walkover <a href="https://beta.giddh.com/" target="_Blank">Giddh.com</a>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/meta/yodlee-success.html:37]: Please revert this

File: apps/web-giddh/src/meta/yodlee-success.html [hunk 11]
```diff
@@ -53,19 +53,19 @@ <h5>You can close this window.</h5>
...
-                apiBaseUrl = 'https://apitest.giddh.com/';
+                apiBaseUrl = 'https://api.giddh.com/';
                 break;
             case 'test.giddh.com':
-                apiBaseUrl = 'https://apitest.giddh.com/';
+                apiBaseUrl = 'https://api.giddh.com/';
                 break;
             case 'stage.giddh.com':
-                apiBaseUrl = 'https://apitest.giddh.com/';
+                apiBaseUrl = 'https://api.giddh.com/';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/meta/yodlee-success.html:62]: apirelease

File: apps/web-giddh/src/meta/yodlee-success.html [hunk 12]
```diff
@@ -53,19 +53,19 @@ <h5>You can close this window.</h5>
         switch (window.location.hostname) {
             case 'localhost':
             case 'dev.giddh.com':
-                apiBaseUrl = 'https://apitest.giddh.com/';
+                apiBaseUrl = 'https://api.giddh.com/';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/meta/yodlee-success.html:56]: apidev

File: apps/web-giddh/src/meta/yodlee-success.html [hunk 13]
```diff
@@ -53,19 +53,19 @@ <h5>You can close this window.</h5>
...
                 break;
             case 'test.giddh.com':
-                apiBaseUrl = 'https://apitest.giddh.com/';
+                apiBaseUrl = 'https://api.giddh.com/';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/meta/yodlee-success.html:59]: apitest

File: apps/web-giddh/webpack.cordova.local.partial.js [hunk 14]
```diff
@@ -10,7 +10,7 @@ module.exports = {
             'errlyticsNeeded': JSON.stringify(false),
             'errlyticsKey': JSON.stringify(''),
             'AppUrl': JSON.stringify('./'),
-            'ApiUrl': JSON.stringify('https://apitest.giddh.com/'),
+            'ApiUrl': JSON.stringify('https://apirelease.giddh.com/'),
```
Comment (prateek-giddh) [inline apps/web-giddh/webpack.cordova.local.partial.js:13]: apidev

File: apps/web-giddh/webpack.local.partial.js [hunk 15]
```diff
@@ -10,7 +10,7 @@ module.exports = {
             'errlyticsNeeded': JSON.stringify(false),
             'errlyticsKey': JSON.stringify(''),
             'AppUrl': JSON.stringify('http://localhost:3000/'),
-            'ApiUrl': JSON.stringify('https://apitest.giddh.com/'),
+            'ApiUrl': JSON.stringify('https://apirelease.giddh.com/'),
```
Comment (prateek-giddh) [inline apps/web-giddh/webpack.local.partial.js:13]: apidev

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts (hunk 4 above)
Comment (ravindersinghdang) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts:38]: Prateek, Doc is already added.

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss [hunk 16]
```diff
@@ -19,6 +19,9 @@
     flex: 1 1 auto;
 }
 
+.total-amt{
```
[the remark this answers is not in the document]
Comment (aditi-tiwari17) [reply apps/web-giddh/src/app/invoice/preview/invoice.preview.component.scss:22]: Yes, this is done in code improvement. Will raise PR for this after a couple of days.

File: apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.scss (hunk 3 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/tb-pl-bs/components/filter/tb-pl-bs-filter.component.scss:99]: Done

File: apps/web-giddh/src/app/gst/gst.component.scss [hunk 17]
```diff
@@ -403,17 +403,26 @@ hr {
 }
 
 .header-info .btn-group > .dropdown-menu {
-    margin-top: 40px;
+    margin-top: 0;
+}
+::ng-deep #monthPicker button.btn.btn-default.btn-info {
+    background: $chetwode-blue !important;
+    color: #fff!important;
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/gst/gst.component.scss:410]: Done

File: apps/web-giddh/src/app/manufacturing/edit/mf.edit.component.html [hunk 18]
```diff
@@ -4,26 +4,26 @@
...
         <button class="btn btn-default pull-right" (click)="goBackToListPage()" type="button">Back</button>
     </div>
 
     <!-- form -->
-    <div class="col-xs-12 mrT2">
-        <div class="row">
-            <form class="mrB2 container pull-left" name="manufaturingForm" #manufaturingForm="ngForm">
+    <div class="col-xs-12 col-lg-8 col-md-12 mr-t2">
+        <div>
+            <form class="mr-b2" name="manufaturingForm" #manufaturingForm="ngForm">
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/manufacturing/edit/mf.edit.component.html:16]: Done

File: apps/web-giddh/src/app/new-vs-old-Invoices/new-vs-old-Invoices.component.html [hunk 19]
```diff
@@ -3,450 +3,232 @@
...
-                            <div *ngIf="SelectMonth.invalid">please select month</div>
+                        <div class="h-25px" *ngIf="SelectMonth.invalid">
+                            <div class="text-danger">please select month</div>
                         </div>
                     </div>
 
-                    <div class="form-group mrR1 max500" *ngIf="selectedType==='quater'">
-                        <label>Select Quarter</label>
-                        <sh-select placeholder="Search Quarter..." [multiple]='false' name="selectedQuater" [(ngModel)]="selectedQuater" #SelectQuater="ngModel" [options]="quaterOptions" required>
+                    <div class="form-group mrR1 max500  d-initial flex-column" *ngIf="selectedType==='quater'">
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/new-vs-old-Invoices/new-vs-old-Invoices.component.html:28]: this is appearing in many places, I can't take the risk for it

File: apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.html [hunk 20]
```diff
@@ -370,6 +368,32 @@
...
+                        <div class="col-sm-12 col-xl-9 col-lg-8 col-md-9 col-xs-12">
+                            <div class="row">
+                            <div class="col-md-10">
+                                <input type="text" class="form-control" placeholder="Select Date" [bsConfig]="{ dateInputFormat: giddhDateFormat }"
+                                    name="purchaseOrderDate" [(ngModel)]="purchaseOrder.voucherDetails.voucherDate" bsDatepicker>
+                            </div>
+                            </div>
+                        </div>
+                    </div>
+                    <div class="row mTB1" *ngIf="isMobileScreen">
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.html:384]: updated

File: apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.scss [hunk 21]
```diff
@@ -910,15 +924,22 @@ span.eligible-ITC {
 }
 
 // card css
+.show-in-mobile {
+    background: #fff;
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.scss:926]: Done

File: apps/web-giddh/src/app/purchase/purchase-order-preview/purchase-order-preview.component.scss [hunk 22]
```diff
@@ -969,4 +985,54 @@ table.tax-table.item {
...
+.loader:before {
+    display: block;
+    position: absolute;
+    z-index: 1000000;
+    content: "";
+    left: 0px;
+    top: 0px;
+    width: 200px;
+    height: 4px;
+    background-color: #5b64c9;
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/purchase/purchase-order-preview/purchase-order-preview.component.scss:1036]: Done

File: apps/web-giddh/src/app/sales/create/sales.invoice.component.scss [hunk 23]
```diff
@@ -501,7 +501,7 @@ textarea:focus {
     top: 120px;
 }
 
-.cashinvoice_tipImg.search-arrow-image {
+.cash-invoice-tip-Img.search-arrow-image {
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/sales/create/sales.invoice.component.scss:504]: Done

File: apps/web-giddh/src/app/settings/permissions/form/form.component.scss [hunk 24]
```diff
@@ -98,21 +102,32 @@ i.icon-cross {
...
+        position: relative;
+        right: 2px
+    }
+}
 @media (max-width: 1199px) {
     .create-btn {
         float: none !important;
         margin-top: 10px;
     }
+    .permissionForm {
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/settings/permissions/form/form.component.scss:116]: Done

File: apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html [hunk 25]
```diff
@@ -145,14 +131,51 @@
     </ng-container>
     <!-- taxes container -->
 </div>
+
+<!-- hinding discount for isDebtorCreditorGroups due to API deployment in-process -->
+<div class="horizontal-form mr1" *ngIf="false">
+    <div class="mb-2 selectBox">
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/shared/header/components/group-update/group-update.component.html:137]: Done

**Review by aditi-tiwari17 — APPROVED** [review]

**Review by shalinee-walkover — APPROVED** [review]
Approve

**Review by prateek-giddh — APPROVED** [review]

**Review by rohit-walkover — APPROVED** [review]

**Pull request merged**